peers/app_request_network.go

```diff
@@ -126,95 +133,155 @@ func NewNetwork(
 		return nil, err
 	}
 
-	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.GetPChainAPI())
+	for _, peer := range manuallyTrackedPeers {
+		logger.Info(
+			"Manually Tracking peer (startup)",
+			zap.String("ID", peer.ID.String()),
+			zap.String("IP", peer.PublicIP.String()),
+		)
+		testNetwork.ManuallyTrack(peer.ID, peer.PublicIP)
+	}
 
-	arNetwork := &appRequestNetwork{
-		network:              testNetwork,
-		handler:              handler,
-		infoAPI:              infoAPI,
-		logger:               logger,
-		lock:                 new(sync.Mutex),
-		validatorClient:      validatorClient,
-		metrics:              metrics,
-		manuallyTrackedPeers: manuallyTrackedPeers,
+	// Connect to a sample of the primary network validators
+	peers, err := infoAPI.Peers(context.Background(), nil)
+	if err != nil {
+		logger.Error(
+			"Failed to get peers",
+			zap.Error(err),
+		)
+		return nil, err
 	}
+	pClient := platformvm.NewClient(cfg.GetPChainAPI().BaseURL)
+	numConnected := 0
+	for _, peer := range peers {
+		vdrs, err := pClient.GetCurrentValidators(context.Background(), constants.PrimaryNetworkID, []ids.NodeID{peer.ID})
+		if err != nil {
+			panic(err)
+		}
+		// Only track the peer if it is a primary network validator
+		if len(vdrs) == 0 {
+			continue
+		}
+		logger.Info(
+			"Manually tracking bootstrap node",
+			zap.String("ID", peer.ID.String()),
+			zap.String("IP", peer.PublicIP.String()),
+		)
+		testNetwork.ManuallyTrack(peer.ID, peer.PublicIP)
+		numConnected++
+		if numConnected >= NumBootstrapNodes {
+			break
+		}
+	}
+
 	go logger.RecoverAndPanic(func() {
 		testNetwork.Dispatch()
 	})
 
+	arNetwork := &appRequestNetwork{
+		network:         testNetwork,
+		handler:         handler,
+		infoAPI:         infoAPI,
+		logger:          logger,
+		lock:            new(sync.RWMutex),
+		validatorClient: validatorClient,
+		metrics:         metrics,
+		trackedSubnets:  trackedSubnets,
+		manager:         manager,
+	}
+
+	arNetwork.startUpdateValidators()
+
 	return arNetwork, nil
 }
 
-// ConnectPeers connects the network to peers with the given nodeIDs.
-// Returns the set of nodeIDs that were successfully connected to.
-func (n *appRequestNetwork) ConnectPeers(nodeIDs set.Set[ids.NodeID]) set.Set[ids.NodeID] {
+func (n *appRequestNetwork) TrackSubnet(subnetID ids.ID) {
+	n.lock.RLock()
+	if n.trackedSubnets.Contains(subnetID) {
+		n.lock.RUnlock()
+		return
+	}
+	n.lock.RUnlock()
```

Comment:
Could pop this out into its own method: 
```go
func (n *appRequestNetwork) containsSubnet(subnetID ids.ID) {
    n.lock.RLock()
    defer n.lock.RUnlock()
    
    return n.trackedSubnets.Contains(subnetID)
}
```
Which cleans up the callsite a little 
```go
func (n *appRequestNetwork) TrackSubnet(subnetID ids.ID) {
    if n.containsSubnet(subnetID) {
        return
    }
    
    // ... 
}
```
🤷 

I always worry that someone's going to make a change that'll cause a deadlock. Might border on paranoia though haha 😅

Reply:
@richardpringle I agree. I think unlocks should always be `defer`'d if possible, so I think putting this in it's own helper is good practice.

Reply:
My personal rule is under 5 lines spanning Lock and non-deferred unlock and no potential errors in between and this passes that test but fine with extracting as well.

Reply:
I went ahead and added the helper. I agree that unlocking with `defer` whenever possible is healthy practice.